cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -379,17 +382,25 @@ protected void onFinish(HttpResponder responder, File uploadedFile) {
           ArtifactSummary artifactSummary = appRequest.getArtifact();
           NamespaceId artifactNamespace =
             ArtifactScope.SYSTEM.equals(artifactSummary.getScope()) ? NamespaceId.SYSTEM : appId.getParent();
-          Id.Artifact artifactId =
-            Id.Artifact.from(artifactNamespace.toId(), artifactSummary.getName(), artifactSummary.getVersion());
 
           KerberosPrincipalId ownerPrincipalId =
             appRequest.getOwnerPrincipal() == null ? null : new KerberosPrincipalId(appRequest.getOwnerPrincipal());
 
           // if we don't null check, it gets serialized to "null"
           String configString = appRequest.getConfig() == null ? null : GSON.toJson(appRequest.getConfig());
-          applicationLifecycleService.deployApp(appId.getParent(), appId.getApplication(), appId.getVersion(),
-                                                artifactId, configString, createProgramTerminator(), ownerPrincipalId,
-                                                appRequest.canUpdateSchedules());
+          if (artifactSummary.isVersionRange()) {
```

Comment:
I dont think I can change deployApp() method to simply take an ArtifactSummary since it is used by other places and it does not contain the namespace info: updateApp() which does some logic with the passed ArtifactSummary and `TrackerAppCreationService` pass the `Id.Artifact` to specify the SYSTEM namespace.